From a18891a9807b209c7c0811e0e6ca08c4570809d0 Mon Sep 17 00:00:00 2001
From: Donn Seeley <donn.seeley@windriver.com>
Date: Tue, 12 Mar 2019 22:00:46 -0700
Subject: [PATCH] Add the glusterfs package

Three of the translator makefiles in glusterfs add unnecessary
-L$(xlatordir) link options.  This option causes the linker to check
the host's $(xlatordir) directory, resulting in contamination that gets
flagged by Yocto's QA rules.

Upstream-Status: Pending

Signed-off-by: Donn Seeley <donn.seeley@windriver.com>
[Xulin: minor adjust to apply the patch]
Signed-off-by: Xuin Sun <xulin.sun@windriver.com>
[Hongzhi: minor adjust to apply the patch]
Signed-off-by: Hongzhi.Song <hongzhi.song@windriver.com>
---
 xlators/mgmt/glusterd/src/Makefile.am    | 2 +-
 xlators/nfs/server/src/Makefile.am       | 2 --
 xlators/system/posix-acl/src/Makefile.am | 2 --
 3 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/xlators/mgmt/glusterd/src/Makefile.am b/xlators/mgmt/glusterd/src/Makefile.am
index aa75344..834c4b5 100644
--- a/xlators/mgmt/glusterd/src/Makefile.am
+++ b/xlators/mgmt/glusterd/src/Makefile.am
@@ -60,7 +60,7 @@ AM_CPPFLAGS = $(GF_CPPFLAGS) -I$(top_srcdir)/libglusterfs/src \
 
 AM_CFLAGS = -Wall $(GF_CFLAGS) $(URCU_CFLAGS) $(URCU_CDS_CFLAGS) $(XML_CFLAGS)
 
-AM_LDFLAGS = -L$(xlatordir) $(URCU_LIBS) $(URCU_CDS_LIBS)
+AM_LDFLAGS = -L$(URCU_LIBS) $(URCU_CDS_LIBS)
 
 CLEANFILES =
 
diff --git a/xlators/nfs/server/src/Makefile.am b/xlators/nfs/server/src/Makefile.am
index 01071a7..297d3cb 100644
--- a/xlators/nfs/server/src/Makefile.am
+++ b/xlators/nfs/server/src/Makefile.am
@@ -31,8 +31,6 @@ AM_CPPFLAGS = $(GF_CPPFLAGS) \
 
 AM_CFLAGS = -Wall $(GF_CFLAGS)
 
-AM_LDFLAGS = -L$(xlatordir)
-
 CLEANFILES =
 
 EXTRA_DIST = nfsserver.sym
diff --git a/xlators/system/posix-acl/src/Makefile.am b/xlators/system/posix-acl/src/Makefile.am
index f57070c..70080a6 100644
--- a/xlators/system/posix-acl/src/Makefile.am
+++ b/xlators/system/posix-acl/src/Makefile.am
@@ -12,8 +12,6 @@ AM_CPPFLAGS = $(GF_CPPFLAGS) -I$(top_srcdir)/libglusterfs/src \
 
 AM_CFLAGS = -Wall $(GF_CFLAGS)
 
-AM_LDFLAGS = -L$(xlatordir)
-
 CLEANFILES =
 
 access-control-compat:
-- 
2.17.1

